pkg/build/controller/controller.go

```diff
@@ -317,7 +317,8 @@ func (bc *BuildPodController) HandlePod(pod *kapi.Pod) error {
 			}
 		}
 	case kapi.PodFailed:
-		build.Status.Reason = ""
+		// NOTE: why do we need to reset the reason for a failed Pod???
+		// build.Status.Reason = ""
```

Comment:
well, this was what was making my changes to the buildobject not to persist.
when the pod enters failed state, they get reset and then the buildobject gets updated with the reset values.